internals-js/src/buildSchema.ts

```diff
@@ -312,41 +341,47 @@ function buildNamedTypeInner(
       for (const fieldNode of definitionNode.fields ?? []) {
         const field = inputObjectType.addField(fieldNode.name.value);
         field.setOfExtension(extension);
-        buildInputFieldDefinitionInner(fieldNode, field);
+        buildInputFieldDefinitionInner(fieldNode, field, errors);
       }
       break;
   }
-  buildAppliedDirectives(definitionNode, type, extension);
+  buildAppliedDirectives(definitionNode, type, errors, extension);
   if (definitionNode.description) {
     type.description = definitionNode.description.value;
   }
   type.sourceAST = definitionNode;
 }
 
-function buildFieldDefinitionInner(fieldNode: FieldDefinitionNode, field: FieldDefinition<any>) {
+function buildFieldDefinitionInner(
+  fieldNode: FieldDefinitionNode,
+  field: FieldDefinition<any>,
+  errors: GraphQLError[],
+) {
   const type = buildTypeReferenceFromAST(fieldNode.type, field.schema());
-  field.type = ensureOutputType(type, field.coordinate, fieldNode);
+  field.type = ensureOutputType(type, field.coordinate, fieldNode, errors);
   for (const inputValueDef of fieldNode.arguments ?? []) {
-    buildArgumentDefinitionInner(inputValueDef, field.addArgument(inputValueDef.name.value));
+    buildArgumentDefinitionInner(inputValueDef, field.addArgument(inputValueDef.name.value), errors);
   }
-  buildAppliedDirectives(fieldNode, field);
+  buildAppliedDirectives(fieldNode, field, errors);
   field.description = fieldNode.description?.value;
   field.sourceAST = fieldNode;
 }
 
-function ensureOutputType(type: Type, what: string, node: ASTNode): OutputType {
+function ensureOutputType(type: Type, what: string, node: ASTNode, errors: GraphQLError[]): OutputType | undefined {
```

Comment:
**observation** Personally, I prefer the `ensureYYYY` methods that throw, since it feels what I'd expect to happen as a caller. I don't love having to add an `error` input to everything. I'd expect the caller to do the error array aggregation, but maybe that's a bit too cumbersome for these cases?

Reply:
I'd preface by saying that I agree this probably won't and shouldn't win any code beauty contest.

That said, pragmatically, I'll point out that it's all private methods and that having throw-catch everywhere would imo be less readable here (and more generally, I feel throw-catch doesn't always land itself super well to error aggregations due to interrupting the control flow while error aggregation is about collecting errors but preserving the normal flow; but of course, in practice, "it depends").

I'd agree though that as we evolve such file, there might come a point where we'd want to refactor things a bit more, but it felt a bit more effort than is worth at this stage. Definitively a judgment call :).

Reply:
That makes sense to me, thanks for writing that out.

I'm starting to see that `try`/`catch` doesn't make sense for these cases. Mostly I wanted to point it out as @queerviolet had the same need to aggregate errors without interrupting control flow. I was hoping we could start to establish some patterns for that case, since I think y'all are solving it in slightly different ways (though everyone is agreeing to stay away from `try`/`catch` 😄 ).